server.py

```
@@ -135,205 +171,375 @@ def format_duration(seconds: int) -> str:
+    cached_value, _, _ = cache.get(cache_key)
+    if cached_value:
+        return cached_value
+
+    api_token = get_api_token()
+    client = TimeCampClient(api_token)
+    
+    # Validate date format
+    try:
+        datetime.strptime(date, "%Y-%m-%d")
+    except ValueError:
+        raise ToolError("Invalid date format. Use YYYY-MM-DD")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)